### ISSUE TITLE
Prediction Outcome of FFM not 0-1 (logistic transform required?)

### ISSUE THREAD

**Issue by leoanghelescu** [Intel-bigdata/imllib-spark#19]
Hi Vincent,

Question at the end about the final transformation (I'm expecting a logit function) of the predict return value:

In FFMExample.scala

val scores: RDD[(Double, Double)] = testing.map(x => {
  val p = ffm.predict(x._2)
  val ret = if (p >= 0.5) 1.0 else -1.0
  (ret, x._1)
})
And the returned value of predict is incremented as such and never transformed (to a 0-1 range)

t += weights(w1_index + d) * weights(w2_index + d) * v

But I do see this logit transformation when we are doing computeFFM

val expnyt = math.exp(-label * t)
val tr_loss = math.log(1 + expnyt)
My main question really is, how do we know that the return value from predict is range (0,1)?

Thanks again for your help!!
Leo

**Comment by VinceShieh** [comment]
Thanks for your comments. Actually, here we are getting the hyperthesis instead of the probability, so users need to calculate the probability based on this value in their test app. basically, we were trying to reuse some code for FFMGradient and FFMModel. Apparently, it makes more sense to return the probability here. PR will be submitted soon, thanks.

**Comment by VinceShieh** [comment]
pls contact us if the problem still exists. Close the issue now
